packages/loader/container-loader/src/test/container.spec.ts

```diff
@@ -50,36 +46,6 @@ class MockContainer extends TypedEventEmitter<IContainerEvents> implements Parti
 }
 
 describe("Container", () => {
-    describe("constructor", () => {
-        const oldRawConfig = sessionStorageConfigProvider.value.getRawConfig;
-        let injectedSettings = {};
-
-        before(() => {
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-return
-            sessionStorageConfigProvider.value.getRawConfig = (name) => injectedSettings[name];
-        });
-
-        afterEach(() => {
-            injectedSettings = {};
-        });
-
-        after(() => {
-            sessionStorageConfigProvider.value.getRawConfig = oldRawConfig;
-        });
-
-        it("Fluid.Container.CatchUpBeforeDeclaringConnected = true, use CatchUpMonitor", () => {
```

Comment:
Why remove this test? Looks like it would still pass?

Reply:
It failed somewhere (maybe in a full PR, i.e. requiring more changes).
This is coming (see https://github.com/microsoft/FluidFramework/pull/12042 for a full set of changes), but we do not need to change Fluid.Container.CatchUpBeforeDeclaringConnected to fully test all this logic and all state permutations, without a need-to-know internal composition of classes and their names.

Basically, once testing for feature gate is moved up the stack (and not buried inside classes), there is zero value in testing that we read feature gate - we are testing really one line in createConnectionStateHandler(), which is not very useful test. For the test we can directly test implementation and its behavior via calling (createConnectionStateHandlerCore(false) & createConnectionStateHandlerCore(true), which is done (already, and more tests coming in above mentioned PR) in connectionStateHandler.spec.ts